main.rb

```diff
@@ -0,0 +1,470 @@
+    end
+    arr
+  end
+
+  # my_each_with_index method definition
+  def my_each_with_index
+    return to_enum :my_each unless block_given?
+
+    arr = self
+    0.upto(arr.length - 1) do |ii|
+      yield arr[ii], ii
+    end
+    arr
+  end
+
+  # my_select method definition
+  def my_select
+    return to_enum :my_select unless block_given?
+
+    arr = self
+    back = []
+    arr.my_each { |x| back << x if yield x }
+    back
+  end
+
+  def param_reg(par, arr_i, t_reg, t_patt, who)
+    # who is false for my_all method call
+    # who is true for my_none method call
+    return [t_reg, t_patt] unless par != ''
+
+    if par.class == Regexp && (!par.match(arr_i).nil? ^ who)
+      t_reg += 1
+    elsif par.class != Regexp && arr_i.is_a?(Module.const_get(par.to_s))
+      t_patt += 1
+    end
+    [t_reg, t_patt]
+  end
+
+  # my_all? method definition
+  def my_all?(param = '')
+    arr = self
+    any_h = { t_b_e: 0, t_e: 0, t_rxp: 0, t_p: 0, who: false }
+
+    0.upto(arr.length - 1) do |i|
+      if param == ''
+        if block_given?
+          any_h[:t_b_e] += 1 if yield arr[i]
+        elsif !arr[i].nil? && arr[i] != false
+          any_h[:t_e] += 1
+        end
+      end
+      any_h[:t_rxp], any_h[:t_p] = param_reg(param, arr[i], any_h[:t_rxp], any_h[:t_p], false)
+    end
+    any_element_true(arr.length, any_h[:t_b_e], any_h[:t_e], any_h[:t_rxp], any_h[:t_p])
+
+    # if we reach this step and all elements are thruty, we exit true.
+  end
+
+  def no_block_count(element_check, block, arr_i)
+    element_check += 1 if (!arr_i.nil? || arr_i != false) && !block
+    element_check
+  end
+
+  # my_any? method definition
+  def my_any?(param = String)
+    arr = self
+    any_h = { true_block_elements: 0, true_elements: 0, t_rxp: 0, true_pattern: 0, who: false }
+
+    0.upto(arr.length - 1) do |i|
+      if block_given?
+        any_h[:true_block_elements] += 1 if yield arr[i]
+      end
+      any_h[:true_block_elements] = no_block_count(any_h[:true_block_elements], block_given?, arr[i])
+      any_h[:t_rxp], any_h[:true_pattern] = param_reg(param, arr[i], any_h[:t_rxp], any_h[:true_pattern], false)
+    end
+    any_h[:true_block_elements] >= 1 || any_h[:t_rxp] >= 1 || any_h[:true_pattern] >= 1
+  end
+
+  def any_element_true(len, fbe, fel, freg, fpa)
+    # method used inside my_none
+    fbe == len || fel == len || freg == len || fpa == len
+  end
+
+  # my_none? method definition
+  def my_none?(param = '')
+    arr = self
+    any_h = { f_b_e: 0, f_e: 0, f_rxp: 0, f_p: 0, who: true }
+
+    0.upto(arr.length - 1) do |i|
+      if param == ''
+        if block_given?
+          any_h[:f_b_e] += 1 unless yield arr[i]
+        elsif arr[i].nil? || arr[i] == false
+          any_h[:f_e] += 1
+        end
+      end
+      any_h[:f_rxp], any_h[:f_p] = param_reg(param, arr[i], any_h[:f_rxp], any_h[:f_p], any_h[:who])
+    end
+    any_element_true(arr.length, any_h[:f_b_e], any_h[:f_e], any_h[:f_rxp], any_h[:f_p])
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when a class is passed as an argument returns true if none of the collection is a member of such class
```ruby
array = Array.new(100) { rand(0...9) }
array.my_none?(String) == true #true
```
- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
```ruby
words = %w[dog door rod blade]
words.my_none?(5) == words.none?(5) #true
```

Reply:
This method is fixed on my end now for the suggested cases:
irb(main):198:0> array = Array.new(100) { rand(0...9) }
=> [0, 0, 8, 6, 7, 7, 0, 8, 3, 8, 6, 8, 3, 0, 4, 5, 0, 3, 6, 5, 8, 2, 3, 4, 4, 7, 3, 4, 5, 6, 0, 7, 5, 8, 7, 6, 3, 8, 4, 6, 8, 5, 6, 3, 4, 7, 6, 6, 4, 0, 2, 3, 7, 7, 2, 1, 1, 8, 6, 3, 4, 8, 7, 3, 5, 3, 5, 4, 2, 1, 4, 0, 7, 2, 6, 4, 0, 3, 8, 4, 6, 3, 3, 2, 8, 2, 8, 2, 5, 0, 4, 7, 2, 3, 4, 6, 4, 4, 2, 7]
irb(main):199:0> array.my_none?(String) == true
=> true
irb(main):200:0> words = %w[dog door rod blade]
=> ["dog", "door", "rod", "blade"]
irb(main):201:0> words.my_none?(5) == words.none?(5)
=> true
irb(main):202:0> words.my_none?(5) 
=> true
irb(main):203:0> words.none?(5)
=> true
irb(main):204:0>